src/controllers/url-controller.ts

```diff
@@ -0,0 +1,153 @@
+import { Request, Response } from 'express';
+import prisma from '../libs/prisma-client';
+
+import { notFoundMsg, successMsg } from '../constants/messages';
+import HTTP_STATUS from '../utils/http-status';
+import logger from '../logs/logger';
+import { convertToHyphenated, generateShortUrl } from '../utils/short-url-generator';
+
+const BASE_URL = process.env.BASE_URL as string;
+
+//  CREATE A SHORTEN URL FROM A LONG URL
+export const createUrl = async (req: Request, res: Response) => {
+  try {
+    const { url, customName } = req.body;
+
+    const existingUrl = await prisma.url.findFirst({
+      where: {
+        originalUrl: url,
+      },
+    });
+    if (existingUrl) {
+      return res.status(HTTP_STATUS.BAD_REQUEST).send({ message: 'Url already exists!' });
+    }
+
+    const urlString = generateShortUrl();
+
+    const custom = customName && convertToHyphenated(customName);
+
+    const originalUrl = url;
+    const shortUrl = custom ? `${BASE_URL}/${custom}` : `${BASE_URL}/${urlString}`;
+    const customNameUrl = custom ? `${custom}` : '';
+
+    const shortenUrl = await prisma.url.create({
+      data: {
+        originalUrl,
+        shortUrl,
+        customName: customNameUrl,
+      },
+    });
+    return res.status(HTTP_STATUS.CREATED).json({ message: successMsg, shortenUrl });
+  } catch (error) {
+    logger.info(error);
```

Comment:
Apply this to all the catch error logger
```suggestion
    logger.error(error);
```

Reply:
Done